subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +511,590 @@ export function _handleReserveDataUpdated(
+    market.id,
+    market.inputToken,
+    event,
+    protocolData
   );
-
-  // udpate market daily / hourly snapshots / financialSnapshots
-  updateSnapshots(
-    protocol,
-    market,
-    borrow.amountUSD,
-    EventType.BORROW,
-    event.block.timestamp,
-    event.block.number
+  const amountUSD = amount.toBigDecimal().times(market.inputTokenPriceUSD);
+  const newBorrowBalances = getBorrowBalances(market, accountID);
+
+  const accountDebtBalance = getOrCreateAccountDebtBalance(market, accountID);
+  accountDebtBalance.sTokenBalance = newBorrowBalances[0];
+  accountDebtBalance.vTokenBalance = newBorrowBalances[1];
+  accountDebtBalance.save();
+
+  manager.createBorrow(
+    asset,
+    accountID,
+    amount,
+    amountUSD,
+    newBorrowBalances[0].plus(newBorrowBalances[1]),
+    market.inputTokenPriceUSD,
+    interestRateType,
+    isIsolated
   );
 }
 
 export function _handleRepay(
   event: ethereum.Event,
   amount: BigInt,
-  marketId: Address,
+  asset: Address,
   protocolData: ProtocolData,
   accountID: Address
 ): void {
-  const market = Market.load(marketId.toHexString());
+  const market = getMarketFromToken(asset, protocolData);
   if (!market) {
-    log.warning("[Repay] Market not found on protocol: {}", [
-      marketId.toHexString(),
+    log.warning("[_handleRepay] Market for token {} not found", [
+      asset.toHexString(),
     ]);
     return;
   }
-  const inputToken = Token.load(market.inputToken);
-  const protocol = getOrCreateLendingProtocol(protocolData);
+  const manager = new DataManager(
+    market.id,
+    market.inputToken,
+    event,
+    protocolData
+  );
+  const amountUSD = amount.toBigDecimal().times(market.inputTokenPriceUSD);
+  const newBorrowBalances = getBorrowBalances(market, accountID);
 
-  // create repay entity
-  const id = `${event.transaction.hash.toHexString()}-${event.logIndex.toString()}`;
-  const repay = new Repay(id);
+  const accountDebtBalance = getOrCreateAccountDebtBalance(market, accountID);
+  let interestRateType: InterestRateType | null = null;
 
-  // get account
-  let account = Account.load(accountID.toHexString());
-  if (!account) {
-    account = createAccount(accountID.toHexString());
-    account.save();
-
-    protocol.cumulativeUniqueUsers += 1;
-    protocol.save();
-  }
-  account.repayCount += 1;
-  account.save();
-
-  const positionId = subtractPosition(
-    protocol,
-    market,
-    account,
-    getBorrowBalance(market, accountID), // try getting balance of account in debt market
-    PositionSide.BORROWER,
-    EventType.REPAY,
-    event
-  );
-  if (positionId === null) {
-    log.warning(
-      "[handleRepay] Position not found for account: {} in transaction; {}",
-      [accountID.toHexString(), event.transaction.hash.toHexString()]
-    );
-    return;
+  // since there may be interest accrued, the balance difference may not exactly
+  // equal repay amount. Here we check whether the balance goes down
+  if (
```

Comment:
I am still trying to figure out if this works?

here is a scenario:
- You borrow 100 USDC variable
- It generates 50USDC in interest, but you have not updated the position ever
- You repay 10 USDC variable
- The new balance would be 140USDC - 100USDC and you would not get an interest rate type

Reply:
good point. I will look for an alternative implementation - probably using `Transfer()` event as you suggested